File: backend/company-service/src/main/java/com/recruitment/platform/company/model/Company.java
package com.recruitment.platform.company.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

import java.time.Instant;

@Entity
@Getter
@Setter
@Table(name = "companies")
public class Company {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private String description;
    private String website;
    private String logoUrl;
    private String companySize;
    private String companyAddress;
    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private CompanyStatus status = CompanyStatus.PENDING;
    private Instant createdAt = Instant.now();

}

File: frontend/recruitment_platform_app/lib/models/company.dart
class Company {
  final int id;
  final String name;
  final String? description;
  final String? website;
  final String? logoUrl;

  Company({
    required this.id,
    required this.name,
    this.description,
    this.website,
    this.logoUrl,
  });

  factory Company.fromJson(Map<String, dynamic> json) {
    return Company(
      id: json['id'],
      name: json['name'],
      description: json['description'],
      website: json['website'],
      logoUrl: json['logoUrl'],
    );
  }
}
